### ISSUE TITLE
"Illegal Character" trying to debug go_test, and passing tests don't show up when run.

### ISSUE THREAD

**Issue by bdleitner** [bazelbuild/intellij#3555]
The problem occurs when run a test in Goland from the "Run/Debug configurations" using bazel test.

If I just run, then the tests run and I can see the results on the console, but the test navigator says "No tests were found" (if they all pass)
![No Tests Found - Goland](https://user-images.githubusercontent.com/7515022/166160166-0681bf32-64a3-4e8d-a576-68e5a5574916.png)
Note that I have "show passing tests" selected, so the passing tests are expected to show up.

Interestingly, if I change one of the tests to fail, I get...
![Failing Test Shows Up](https://user-images.githubusercontent.com/7515022/166160244-be29538d-d55a-48cd-a94e-007d7ea13ade.png)
so the failing test shows up and the passing ones do not.

But the original issue I had is that I'm not able to debug.  If I click the debug button I get:
![Illegal Character](https://user-images.githubusercontent.com/7515022/166160833-e8e7c1b3-465b-4064-830f-e9dae653d72f.png)

I tried upgrading the bazel plugin (and Goland) since I saw that fixing debugging with `go_test` was on the list of recent fixes, but the issue persists after all of that, plus doing `bazel clean` and even nuking my output-user-root directory to force everything to rebuild.

Also note that the illegal character remains at index 116 even if I change the test to have everything commented out, leaving only one empty test method.

Throughout all of this, `bazel build` and `bazel test` continue to work from the terminal as expected.

Versions:
Goland: 2022.1
Bazelisk: 1.11.0
Bazel: 5.1.1
Bazel Plugin: 2022.04.28.0.0-api-version-221

**Comment by bdleitner** [comment]
For a comparison, I also checked with IntelliJ for my java tests (same bazel plugin version) and it worked fine there, so it does seem to be golang specific.

**Comment by bdleitner** [comment]
Further note: I tried adding a `.bazelversion` file to use version 5.0.0, as that version is noted in the bazel plugin.  This dit not help, I got the same error.

**Comment by sgowroji** [comment]
Hello @bdleitner, Could you please respond if you are looking support for the above  issue and also provide a sample code to verify the above issue. Thanks!

**Comment by bdleitner** [comment]
Sorry for the delayed response.  Yes, I'm still looking for support if it's
still happening.  I didn't have a chance to test it out this weekend.
I will try and repro this evening or next after seeing if there are any
updates and I'll post a repo to github with a minimal reproducible example.

But when last I checked it was happening for all tests I tried to debug.
Literally a dummy setup of "here's a method that returns 1" and "here's a
test that verifies it returns 1" (to pass) or "here's a method that
verifies it returns 2" (to fail) had the problem.

**Comment by bdleitner** [comment]
Rechecked with updated Bazel plugin and confirmed the issue still occurs.  
Created https://github.com/bdleitner/illegal_char_demo to demo, with code+screenshots.

**Comment by bdleitner** [comment]
When I updated he plugin to the 08.29- version, I could no longer reproduce this.  It appears this has been fixed.

**Comment by bdleitner** [comment]
I just ran into this again in Goland 2024.2.1.1 with bazel plugin version 2024.08.27.0.1-api-version-242

**Comment by tpasternak** [comment]
Thank you for reporting! @agluszak @jastice can you PTAL? It seems to be a regression

**Comment by tpasternak** [comment]
@bdleitner sorry, I was OOO last week, do you use testify, by any chance?

**Comment by bdleitner** [comment]
On Wed, Sep 25, 2024 at 5:03 AM Tomasz Pasternak ***@***.***>
wrote:

> @bdleitner <https://github.com/bdleitner> sorry, I was OOO last week, do
> you use testify, by any chance?
>

I do not (not familiar with it).

**Comment by bdleitner** [comment]
Note sure if it's related, I tried doing "run test with coverage" and got:

```
FAIL: //go/expr/scope:scope_test (see D:/_bazel_out/wr4qdbwt/execroot/_main/bazel-out/x64_windows-fastbuild/testlogs/go/expr/scope/scope_test/test.log)
INFO: From Testing //go/expr/scope:scope_test:
==================== Test output for //go/expr/scope:scope_test:
ERROR(tools/test/windows/tw.cc:1302) ERROR: src/main/native/windows/process.cc(202): CreateProcessW("D:\_bazel_out\wr4qdbwt\execroot\_main\bazel-out\x64_windows-fastbuild\bin\go\expr\scope\scope_test_\scope_test.exe.runfiles\_main\external\bazel_tools\tools\test\collect_coverage.sh"  go/expr/scope/scope_test_/scope_test.exe): The system cannot find the file specified.
 (error: 2)
ERROR(tools/test/windows/tw.cc:1479) Failed to start test process (arg: D:\_bazel_out\wr4qdbwt\execroot\_main\bazel-out\x64_windows-fastbuild\bin\go\expr\scope\scope_test_\scope_test.exe.runfiles\_main\external\bazel_tools\tools\test\collect_coverage.sh)
================================================================================
ERROR: C:/projects/go/monolith/go/expr/scope/BUILD:10:8: output 'go/expr/scope/scope_test/coverage.dat' was not created
ERROR: C:/projects/go/monolith/go/expr/scope/BUILD:10:8: Testing //go/expr/scope:scope_test failed: not all outputs were created or valid
Target //go/expr/scope:scope_test up-to-date:
  bazel-bin/go/expr/scope/scope_test_/scope_test.exe
INFO: Elapsed time: 0.652s, Critical Path: 0.07s
INFO: 2 processes: 2 local.
ERROR: Build did NOT complete successfully
FAILED: 
//go/expr/scope:scope_test                                               FAILED in 0.0s
  D:/_bazel_out/wr4qdbwt/execroot/_main/bazel-out/x64_windows-fastbuild/testlogs/go/expr/scope/scope_test/test.log

Executed 1 out of 1 test: 1 fails locally.
INFO: Build Event Protocol files produced successfully.

exited with error code 1
Executing tests from //go/expr/scope:scope_test
-----------------------------------------------------------------------------
ERROR(tools/test/windows/tw.cc:1302) ERROR: src/main/native/windows/process.cc(202): CreateProcessW("D:\_bazel_out\wr4qdbwt\execroot\_main\bazel-out\x64_windows-fastbuild\bin\go\expr\scope\scope_test_\scope_test.exe.runfiles\_main\external\bazel_tools\tools\test\collect_coverage.sh"  go/expr/scope/scope_test_/scope_test.exe): The system cannot find the file specified.
 (error: 2)
ERROR(tools/test/windows/tw.cc:1479) Failed to start test process (arg: D:\_bazel_out\wr4qdbwt\execroot\_main\bazel-out\x64_windows-fastbuild\bin\go\expr\scope\scope_test_\scope_test.exe.runfiles\_main\external\bazel_tools\tools\test\collect_coverage.sh)
```

**Comment by tpasternak** [comment]
Honestly I think there is an error in the rules you use, and therefore no tests are actually run

**Comment by tpasternak** [comment]
Seems that one of the actions declares an output, but fails to create it

**Comment by bdleitner** [comment]
Yeah, run with coverage doesn't seem to work because the coverage isn't generated. Seems unrelated to the illegal char issue.

**Comment by tpasternak** [comment]
can I close the ticket then?

**Comment by bdleitner** [comment]
The thing about coverage not working was an aside that I thought at first
might be related. It seems not to be, but the last time I tried anything
the illegal character issue on trying to debug tests was still active.

I should have kept the coverage thing separate; sorry for the confusion.

**Comment by tpasternak** [comment]
So may I ask you to open a new issue with a repro?

**Comment by bdleitner** [comment]
Wait... Is the illegal character issue fixed? Do you know when the fix will
go out?

I mentioned another problem I ran into on this bug, thinking at first it
might be connected. As it turns out, it's not. I can file a separate issue
for that with more information for a repro, bit that has nothing to do with
the illegal character bug, so unless that issue is fixed, this should stay
open.

**Comment by tpasternak** [comment]
oh, misunderstood you

**Comment by bdleitner** [comment]
I've updated my repro at https://github.com/bdleitner/illegal_char_demo to use bzlmod like my other big repo where I'm having the issue, and updated the screenshots.

**Comment by iliakondratev** [comment]
Issue reproduces in Goland 2024.2.1.1 on Windows but not on macOS.

**Comment by iliakondratev** [comment]
<img width="1073" alt="image" src="https://github.com/user-attachments/assets/ed0c740c-65f9-438d-8b24-cbc99bfbc2a2">

Debugging the given example as `go_binary` on Windows isn't functioning too, showing a different error message

**Comment by tpasternak** [comment]
Fixed in #6938
